airbyte-integrations/connectors/destination-bigquery/src/main/kotlin/io/airbyte/integrations/destination/bigquery/formatter/BigDecimalUtil.kt

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright (c) 2025 Airbyte, Inc., all rights reserved.
+ */
+
+package io.airbyte.integrations.destination.bigquery.formatter
+
+import java.math.BigDecimal
+import java.math.BigInteger
+
+object BigDecimalUtil {
+    /** Returns the largest representable BigDecimal for a given (precision, scale) pair. */
+    fun maxForRange(precision: Int, scale: Int) =
+        BigDecimal(BigInteger("9".repeat(precision)), scale)
+}
+
+/**
```

Comment:
I didn't end up using these new functions, but left them in since they're not costing anything.